subsys/net/ip/net_tc.c

```diff
@@ -71,7 +71,9 @@ enum net_verdict net_tc_submit_to_tx_queue(uint8_t tc, struct net_pkt *pkt)
 	net_pkt_set_tx_stats_tick(pkt, k_cycle_get_32());
 
 #if NET_TC_TX_EFFECTIVE_COUNT > 1
-	if (k_sem_take(&tx_classes[tc].fifo_slot, K_FOREVER) != 0) {
+	k_timeout_t timeout = k_is_in_isr() ? K_NO_WAIT : K_FOREVER;
```

Comment:
No objections for this, but I thought that for ISR context, the timeout is automatically set to `K_NO_WAIT` by the relevant kernel API.